engine/runtime/src/main/scala/org/enso/interpreter/instrument/execution/JobProcessor.scala

```diff
@@ -15,8 +15,14 @@ trait JobProcessor {
     */
   def run[A](job: Job[A]): Future[A]
 
-  /** Stops the job processor.
+  /** Runs a job with the provided context in the background.
+    *
+    * @param job a job to execute
+    * @return the future result of an asynchronous computation
     */
+  def runBackground[A](job: Job[A]): Future[A]
```

Comment:
Just a slightly unrelated _side note_: I am not much knowledgeable about [Scala Future](https://www.scala-lang.org/api/2.13.x/scala/concurrent/Future.html), but it seems to me, it might be enough to just return:
```
def backgroundExecutor(): ExecutionContext
```
and just create the `Future` myself without specifying a `Job` at all. However I may be missing some motivation why the `Job` abstraction exists...